rpc/types/src/macros.rs

```diff
@@ -0,0 +1,277 @@
+
+        // The base `struct` name.
+        $type_name:ident,
+
+        // The request type (and any doc comments, derives, etc).
+        $( #[$request_type_attr:meta] )*
+        $request_base_type:ty {
+            // And any fields.
+            $(
+                $( #[$request_field_attr:meta] )*
+                $request_field:ident: $request_field_type:ty,
+            )*
+        },
+
+        // The response type (and any doc comments, derives, etc).
+        $( #[$response_type_attr:meta] )*
+        $response_base_type:ty {
+            // And any fields.
+            $(
+                $( #[$response_field_attr:meta] )*
+                $response_field:ident: $response_field_type:ty,
+            )*
+        }
+    ) => { paste::paste! {
+        #[allow(dead_code)]
+        #[allow(missing_docs)]
+        #[derive(serde::Serialize, serde::Deserialize)]
+        #[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        $( #[$request_type_attr] )*
+        #[doc = $crate::macros::define_request_and_response_doc!(
+            "response",
+            $monero_daemon_rpc_doc_link,
+            $monero_code_commit,
+            $monero_code_filename,
+            $monero_code_filename_extension,
+            $monero_code_line_start,
+            $monero_code_line_end,
+            [<$type_name Request>],
+        )]
+        pub struct [<$type_name Request>] {
+            #[serde(flatten)]
+            pub base: $request_base_type,
+
+            $(
+                $( #[$request_field_attr] )*
+                pub $request_field: $request_field_type,
+            )*
+        }
+
+        ::cuprate_epee_encoding::epee_object! {
+            [<$type_name Request>],
+            $(
+                $request_field: $request_field_type,
+            )*
+            !flatten: base: $request_base_type,
+        }
+
+        #[allow(dead_code)]
+        #[allow(missing_docs)]
+        #[derive(serde::Serialize, serde::Deserialize)]
+        #[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        $( #[$response_type_attr] )*
+        #[doc = $crate::macros::define_request_and_response_doc!(
+            "request",
+            $monero_daemon_rpc_doc_link,
+            $monero_code_commit,
+            $monero_code_filename,
+            $monero_code_filename_extension,
+            $monero_code_line_start,
+            $monero_code_line_end,
+            [<$type_name Response>],
+        )]
+        pub struct [<$type_name Response>] {
+            #[serde(flatten)]
+            pub base: $response_base_type,
+
+            $(
+                $( #[$response_field_attr] )*
+                pub $response_field: $response_field_type,
+            )*
+        }
+
+        ::cuprate_epee_encoding::epee_object! {
+            [<$type_name Response>],
+            $(
+                $response_field: $response_field_type,
+            )*
+            !flatten: base: $response_base_type,
+        }
```

Comment:
## Type generator macro 3/3

Every request/response type now implements `EpeeObject`. Does this look correct?

Reply:
yeah it does